target_benchmark/tasks/AbsTask.py

```diff
@@ -223,37 +305,59 @@ def task_run(
         logger.info(f"start task {self.task_name}")
 
         for dataset_name, dataset_loader in dataset_loaders.items():
+            # construct the path to persistence files
+            path_to_retrieval_results = construct_persistence_path(path_to_retrieval_results_dir, dataset_name, top_k)
+            path_to_downstream_results = construct_persistence_path(path_to_downstream_results_dir, dataset_name, top_k)
+
+            # construct generators
+            prev_retrieval_res_gen = generate_batches_from_file(
+                path_to_retrieval_results,
+                batch_size,
+                RetrievalResultDataModel,
+            )
+            prev_downstream_res_gen = generate_batches_from_file(
+                path_to_downstream_results,
+                batch_size,
+                DownstreamGeneratedResultDataModel,
+            )
+
             logger.info(f"running task on dataset {dataset_name}")
+
             table_id_to_table = dataset_loader.get_table_id_to_table()
+            # some retrieval metrics to track
             total_process_duration = 0
             total_wall_clock_duration = 0
+            total_num_retrieved = 0
+
+            # set up progress bar
             total_num_queries = dataset_loader.get_queries_size()
             progress_bar = tqdm(total=total_num_queries, desc=f"Retrieving Tables for {dataset_name}...")
-            for query_batch in dataset_loader.get_queries_for_task(batch_size):
-                (
-                    retrieval_results,
-                    process_duration,
-                    wall_clock_duration,
-                ) = self._get_retrieval_results(
+            for query_batch in dataset_loader.get_queries_for_task(batch_size=batch_size):
```

Comment:
Can you walk me through why we're not passing a start_idx argument here, and instead pass a generator to _run_retrieval_batch?

It seems that given the patterns you've established with start_idx, we shouldn't even need to hit _run_retrieval_batch if we have results from retrieval_results_dir?

Reply:
ah yes. i realized even with results from `retrieval_results_dir`, we *still* need to read through those files, so we can get the previous results for **calculating the metrics**. so instead of starting with a `start_idx`, I consolidated the logic for retrieval -- whether that be reading the retrieval results or getting new results from the retriever.